addon/components/attach-popover.js

```diff
@@ -479,6 +488,21 @@ export default Component.extend({
     if (hideOn.indexOf('clickout') !== -1) {
       this._hideListenersOnDocumentByEvent.click = this._hideOnClickOut;
       document.addEventListener('click', this._hideOnClickOut);
+
+      // On iOS, `document.addEventListener('click', handler)` doesn't work,
+      // so simulating click events using touch events.
+      if (navigator.userAgent && navigator.userAgent.match(/iPhone|iPad|iPod/)) {
+        this._hideListenersOnDocumentByEvent.touchstart
+          = this._enableDocumentClickSimulationOnClickOutOnIos;
+        document.addEventListener('touchstart', this._enableDocumentClickSimulationOnClickOutOnIos);
+
+        this._hideListenersOnDocumentByEvent.touchmove
+          = this._disableDocumentClickSimulationOnClickOutOnIos;
+        document.addEventListener('touchmove', this._disableDocumentClickSimulationOnClickOutOnIos);
+
+        this._hideListenersOnDocumentByEvent.touchend = this._hideOnClickOutOnIos;
+        document.addEventListener('touchend', this._hideOnClickOutOnIos);
```

Comment:
I'm positive your idea https://github.com/kybishop/ember-attacher/pull/115#discussion_r173999972 is the simplest solution!

But, what I'm worried about and why I've used these three events are that using only `touchend` will change the current behavior of ember-attacher.
Now, on Android devices, 

1. users can click a button then a tooltip will appear
2. users can scroll the screen with the tooltip being visible

because click events won't be triggered when users are scrolling their screen.

However, `touchend` events always happen when users lift their fingers off the screen.
Then, users won't be able to scroll with the tooltip being visible.

If you don't mind it, I think https://github.com/kybishop/ember-attacher/pull/115#discussion_r173999972 can be the best!
And I'd like to update the PR following this idea.

How do you feel?

Reply:
This makes sense. Perhaps we should use `touchstart` instead of `touchend` to share the same behavior of hiding tooltips on scroll?

I'll leave the decision up to you. We can always change it if people think it should be one way or the other.

Reply:
Thanks! I'll try to keep the current behavior, but will remove the devise specific code following your idea!

Reply:
> This makes sense. Perhaps we should use touchstart instead of touchend to share the same behavior of hiding tooltips on scroll?

Not to change the current behavior, we should use all three events to simulate "a click", I think...

* `touchstart` - a user puts their finger on the screen
* `touchmove` - a user moves the finger
* `touchend` - a user lifts the finger off

Some try to simulate click events "perfectly" then they're tracking the finger position (e.g. If the finger moved only within Npx, this should be regarded as a click), but it may be too much...

Currently, if users don't move their fingers from beginning to end, I regard it as a click.

Reply:
Sounds nice enough! I'll update my PR!